### PR TITLE
Add performance roadmap items

### PR DESCRIPTION
**Pull request by satnam6502**
Transcribe the performance improvement work items in the placeholder PR #2931 to the roadmap document. @roberthbailey @bgrant0607

### CODE REVIEW

File: docs/roadmap.md [hunk 1]
```diff
@@ -149,3 +149,58 @@ Allow the client to be authenticated to multiple clusters (https://github.com/Go
 * Issue [#3138](https://github.com/GoogleCloudPlatform/kubernetes/issues/3138) [davidopp] A way to record a real workload and replay it deterministically
 * Issue [#3139](https://github.com/GoogleCloudPlatform/kubernetes/issues/3139) [davidopp] A way to generate a synthetic workload and play it
 * Issue [#2852](https://github.com/GoogleCloudPlatform/kubernetes/issues/2852) and Issue [#3067](https://github.com/GoogleCloudPlatform/kubernetes/issues/3067) [vishnuk] Protect system services against kernel OOM kills and resource starvation. 
+
+# Performance
+Currently we are conflating performance of any size of Kubernetes cluster with scalability. Later we may wish to tease apart these two concerns. As part of overall performance, we also consider the performance of the build and test processes. The main current pain point is the we have no systematic performance measurement mechanism or process.
+
+The goals of the performance related activities are:
+* A collection of performance regression tests that measure the effect of code changes.
+* A dashboard or some other form of storeing and querying historical performance information.
```
Comment (roberthbailey) [inline docs/roadmap.md]: storing

File: docs/roadmap.md [hunk 2]
```diff
@@ -149,3 +149,58 @@ Allow the client to be authenticated to multiple clusters (https://github.com/Go
...
+
+Things we could measure:
+
+1. Time to build from source.
+2. Time to run each test in the end to end suite (or a mean plus standard deviation).
+3. Time taken to perform API operations e.g.
+ 1. List pods, for a varying number of pods (no labels).
+ 2. Time taken to create and tear down clusters of various sizes.
+ 3. Time taken to perform API operations, e.g.
```
Comment (roberthbailey) [inline docs/roadmap.md]: duplicate paste here (lines 167-168 == 164-165)

**Comment by roberthbailey** [comment]
A couple of nits, otherwise LGTM.

**Comment by satnam6502** [comment]
Thanks: I think I fixed the mistakes.

**Comment by roberthbailey** [comment]
I'll wait for Brian to take a look before merging.

**Comment by bgrant0607** [comment]
Thanks. LGTM.

**Pull request merged**